quickwit/quickwit-storage/src/object_storage/s3_compatible_storage.rs

```diff
@@ -127,7 +128,14 @@ fn get_region(s3_storage_config: &S3StorageConfig) -> Option<Region> {
     })
 }
 
-async fn create_s3_client(s3_storage_config: &S3StorageConfig) -> S3Client {
+async fn create_s3_client(s3_storage_config: &S3StorageConfig) -> Arc<S3Client> {
+    static MEMOIZED_CLIENT: Lazy<Mutex<HashMap<S3StorageConfig, Arc<S3Client>>>> =
+        Lazy::new(|| Mutex::new(HashMap::new()));
```

Comment:
I think it would be more explicit if the `S3Client` was mapped to the config object in the `S3CompatibleObjectStorageFactory`